src/types/object.js

```diff
@@ -35,4 +35,12 @@ export default parameterized(T => class ObjectType {
   delete(name) {
     return filter(({ key }) => key !== name, valueOf(this));
   }
+
+  map(fn) {
+    return map(value => valueOf(fn(create(T, value))), valueOf(this));
+  }
+
+  filter(fn) {
+    return filter(({ key }) => fn(sourceOf(this[key]), key), valueOf(this));
```

Comment:
By the same token, I don't think we need `sourceOf` here, and instead can opt to just create a one-off filter context.

Reply:
Sounds good. I fixed that and took out the import for `sourceOf`